mapscript/java/CMakeLists.txt

```
@@ -14,6 +14,9 @@ include_directories(${PROJECT_SOURCE_DIR}/mapscript/java)
 set (CMAKE_SWIG_OUTDIR "${CMAKE_CURRENT_BINARY_DIR}/edu/umn/gis/mapscript")
 set(CMAKE_SWIG_FLAGS -package edu.umn.gis.mapscript)
 
+# hide warnings
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
```

This is a gcc specific switch. Should probably be protected by (untested)
```suggestion
if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
endif()
```

---

Should CMAKE_COMPILER_IS_GNUCXX also be checked for? 
I've added the same approach as https://github.com/mapserver/mapserver/blob/master/CMakeLists.txt#L188

---

Testing CMAKE_COMPILER_IS_GNUCXX is for the C++ compiler, so not relevant when setting CMAKE_C_FLAGS
But thinking about that the test for clang with CMAKE_CXX_COMPILER_ID should be done with CMAKE_C_COMPILER_ID